platform/tests/TESTS/mbed_functional/callback/main.cpp

```diff
@@ -884,10 +886,10 @@ Case cases[] = {
     Case("Testing callbacks with 2 uint64s", test_dispatch2<uint64_t>),
     Case("Testing callbacks with 3 uint64s", test_dispatch3<uint64_t>),
     Case("Testing callbacks with 4 uint64s", test_dispatch4<uint64_t>),
-// IAR currently crashes at link time with this test - skip it as it's well beyond anything needed by real code
-#ifndef __ICCARM__
+#if !defined(__ICCARM__) && defined(MBED_ROM_SIZE) && (MBED_ROM_SIZE >= TEST_MIN_REQ_ROM_SIZE)
```

Comment:
This looks like if `MBED_ROM_SIZE` is not defined, the test case will not be included. Don't we want to still test even if `MBED_ROM_SIZE` is not defined? This way, we don't skip the test on accident. We ensure if a small target wants to skip the test, it has to define `MBED_ROM_SIZE` (like it should do anyway).

We should include this test case if `MBED_ROM_SIZE` is not defined, or if `MBED_ROM_SIZE` is defined and there is enough room in flash.

```diff
- #if !defined(__ICCARM__) && defined(MBED_ROM_SIZE) && (MBED_ROM_SIZE >= TEST_MIN_REQ_ROM_SIZE)
+ #if !defined(__ICCARM__)
+ #if !defined(MBED_ROM_SIZE) || (defined(MBED_ROM_SIZE) && (MBED_ROM_SIZE >= TEST_MIN_REQ_ROM_SIZE))
```

Reply:
I thought about that to include the test if MBED_ROM_SIZE is not defined but if the target doesn't have enough ROM and not defined ROM size will have problem.

This callback_big  test `test_dispathx` template function repeatedly doing the same set of function call across all test_dispatch 1,2,3,4,5.

Reply:
If the target doesn't have enough ROM, it's better that it fails and the programmer add `MBED_ROM_SIZE` to skip the test. As currently written in this PR, if a target does have enough flash, but doesn't define `MBED_ROM_SIZE`, we'll silently skip the test, missing out on test coverage silently.